cosine_matcher.py

```diff
@@ -27,54 +25,67 @@ def __init__(self, encoding='utf-8', analyzer='word', ngram_range=(1,1), \
 
         This code assumes that you have already cleaned & processed the text corpus. 
 
-        '''
+        SAMPLE USAGE:
+        searcher = CosineMatcher()
+        searcher.train('C:/Users/User/Desktop/myFile.csv')
+        searcher.find('manila irrigation')
+        >>> {'manila north rd ...
+        """
         self.match_corpus = None
         self.matrix = None 
         self.vectorizer = TfidfVectorizer(encoding=encoding, analyzer=analyzer,\
             ngram_range=ngram_range, min_df=min_df, max_df=max_df,\
             use_idf=use_idf)
 
 
-    def train(self, corpus, train_on='training_col'):
-        '''
+    def train(self, corpus, train_on='searchspace'):
+        """
         Fit the training corpus to the TF-IDF Vectorizer.
 
         corpus: Path to CSV file containing the training corpus.
         train_on: Name of the column in the CSV.
-        '''
-        df_corpus = pd.read_csv(corpus)
-        self.match_corpus = df_corpus[
-            df_corpus[train_on].isnull()==False].reset_index()
+        
+        SAMPLE USAGE
+        searcher = CosineMatcher()
+        searcher.train('C:/Users/User/Desktop/myFile.csv')
+        """
+        if type(df_corpus) in [str, unicode]:
+            corpus = pd.read_csv(corpus)
```

Comment:
CSV processing function is missing! add as separate function. 

Then once we have that, get rid of `read_csv` here b/c `corpus` will always be a dataframe generated by that missing csv processing function.